src/Bicep.Core.UnitTests/Diagnostics/LinterRuleTests/UseRecentApiVersionRule_InReferenceFunctions_Tests.cs

```diff
@@ -29,16 +33,30 @@ static InReferenceFunctions()
             {
                 // Test with the linter using the fake resource types from FakeResourceTypes (to guard against failures due to Azure changes)
                 // Note: The compiler does not know about these fake types, only the linter.
-                apiVersionProvider = new ApiVersionProvider(BicepTestConstants.Features, BicepTestConstants.AzResourceTypeLoader);
-                apiVersionProvider.InjectTypeReferences(ResourceScope.ResourceGroup, FakeResourceTypes.GetFakeResourceTypeReferences(FakeResourceTypes.ResourceScopeTypes));
+                apiVersionProvider = new ApiVersionProvider(
+                    BicepTestConstants.Features,
+                    Enumerable.Empty<ResourceTypeReference>());
+                apiVersionProvider.InjectTypeReferences(
+                    ResourceScope.ResourceGroup,
+                    FakeResourceTypes.GetFakeResourceTypeReferences(FakeResourceTypes.ResourceScopeTypes));
             }
 
             private static void TestGetFunctionCallInfo(string bicep, string expectedFunctionCall, string? expectedResourceType, string? expectedApiVerion)
             {
                 ExpectedFunctionInfo typedExpected = new(expectedFunctionCall, expectedResourceType, expectedApiVerion);
+                var fakeResourceTypeReferences = FakeResourceTypes.GetFakeResourceTypeReferences(FakeResourceTypes.ResourceScopeTypes);
```

Comment:
Can this be shared with similar code elsewhere?

Reply:
what do you mean? I am sorry :( I don't follow